### PR TITLE
MTU fix for cmd-forwarder-vpp#557

### PR DESCRIPTION
**Pull request by edwarnicke**
Signed-off-by: Ed Warnicke <hagbard@gmail.com>

### CODE REVIEW

File: pkg/networkservice/connectioncontext/mtu/common.go [hunk 1]
```diff
@@ -34,36 +34,7 @@ const (
...
-	log.FromContext(ctx).
-		WithField("SwIfIndex", setMTU.SwIfIndex).
-		WithField("MTU", setMTU.Mtu).
-		WithField("duration", time.Since(now)).
-		WithField("vppapi", "HwInterfaceSetMtu").Debug("completed")
-	return nil
-}
-
-func setVPPL3MTU(ctx context.Context, conn *networkservice.Connection, vppConn api.Connection, isClient bool) error {
+func setVPPMTU(ctx context.Context, conn *networkservice.Connection, vppConn api.Connection, isClient bool) error {
```
Comment (edwarnicke) [inline pkg/networkservice/connectioncontext/mtu/common.go:37]: Abandon setting the LinkMTU entirely as it no longer works remotely as expected.

This appears to be due to changes in VPP driven by changes in DPDK on MTU handling no longer following 802.3 on MTU handling.

File: pkg/networkservice/mechanisms/vxlan/mtu/common.go [hunk 2]
```diff
@@ -67,10 +63,29 @@ func getMTU(ctx context.Context, vppConn api.Connection, tunnelIP net.IP) (uint3
...
+		// total - 54 bytes
+		return 54
+	}
+	// outer ipv6 header - 40 bytes
+	// outer udp header - 8 bytes
+	// vxlan header - 8 bytes
+	// inner ethernet header - 14 bytes
+	// optional overhead for 802.1q vlan tags - 4 bytes
+	// total - 74 bytes
+	return 74
```
Comment (edwarnicke) [inline pkg/networkservice/mechanisms/vxlan/mtu/common.go:90]: Correctly handle overhead for VXLAN (previously we'd only been handling for IPv4 with no available overhead for 802.1q tags).

File: pkg/networkservice/mechanisms/wireguard/mtu/common.go [hunk 3]
```diff
@@ -67,10 +63,32 @@ func getMTU(ctx context.Context, vppConn api.Connection, tunnelIP net.IP) (uint3
...
+		return 60
+	}
+	//  40-byte outer IPv4 header
+	//  8-byte outer UDP header
+	//  4-byte type
+	//  4-byte key index
+	//  8-byte nonce
+	// 16-byte authentication tag
+	// 80 bytes total
+	return 80
```
Comment (edwarnicke) [inline pkg/networkservice/mechanisms/wireguard/mtu/common.go:93]: Correctly handle overhead for IPv4 case for wireguard.

**Review by denis-tingaikin — APPROVED** [review]
All changes are looking good. 


@glazychev-art I think it'd be interested to look at these changes for you.

**Pull request merged**